### PR TITLE
Refactor: Remove devise_invitable and simplify invitation workflow

### PR DESCRIPTION
**Pull request by koetsier**
### What
- Removed `devise_invitable` gem from the application.
- Simplified the invitation process by eliminating the need to invite both the `User` model and the `Membership` model.
  - Invitations are now handled solely through the `Membership` model.
- This refactor addresses several issues:
  - Reduced complexity in the invitation workflow, making it easier to maintain and understand.
  - Fixed edge cases where duplicate or inconsistent invitations occurred due to the dual-model approach.
  - Streamlined user creation and association, ensuring a smoother onboarding experience.
- Updated relevant tests and documentation to reflect the new workflow.
- Added request and form model tests

### Why
This change simplifies the codebase and enhances the reliability of the invitation system while maintaining all existing functionality.

Link to JIRA card (if applicable):
https://technologyprogramme.atlassian.net/browse/GW-1986

### CODE REVIEW

File: app/controllers/memberships_controller.rb [hunk 1]
```diff
@@ -4,39 +4,7 @@ class MembershipsController < ApplicationController
...
-        text: "View only",
-        hint: <<~HINT.html_safe,
-          <span>View locations and IPs, team members, and logs</span><br>
-          <span>Cannot manage locations and IPs</span><br>
-          <span>Cannot add or remove team members</span>
-        HINT
-      ),
-    ]
-  end
+  def edit; end
```
Comment (leonaAtkins) [inline app/controllers/memberships_controller.rb:7]: does this break something if you remove the method altogether?

File: config/locales/en.yml [hunk 2]
```diff
@@ -5,6 +5,15 @@ en:
...
     errors:
       models:
+        accept_invitation_form:
+          attributes:
+            name:
+              blank: "Name can't be blank"
+            password:
+              weak_password: "Password is not strong enough. Choose a different password."
+              blank: "Password can't be blank"
+              too_short: "Password is too short (minimum is 6 characters)"
```
Comment (leonaAtkins) [inline config/locales/en.yml:15]: Again thing minimum should be 8 chars.

**Comment by leonaAtkins** [comment]
@koetsier is this in a 'deployable' state, and have you deployed it to Dev, if not, any objections If I deploy it to Dev / Alpaca, so we could test it?

File: app/controllers/memberships_controller.rb (hunk 1 above)
Comment (koetsier) [reply app/controllers/memberships_controller.rb:7]: No, we could remove it altogether

Comment (koetsier) [reply app/controllers/memberships_controller.rb:7]: yes we could remove the method

File: app/models/accept_invitation_form.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  include ActiveModel::Validations
+  include Devise::Models::Zxcvbnable
+
+  attr_accessor :name, :password, :invitation_token
+
+  validates :name, presence: true, if: :requires_validation
+
+  validates :password,
+            presence: true,
+            length: { within: 6..80 },
```
[the remark this answers is not in the document]
Comment (koetsier) [reply app/models/accept_invitation_form.rb:14]: 👍

**Comment by leonaAtkins** [comment]
Sjo I deployed this to Dev and it failed the smoke tests, you can take a look at the report here.
https://eu-west-2.console.aws.amazon.com/codesuite/codebuild/269352317959/projects/govwifi-smoke-tests/build/govwifi-smoke-tests%3A93068b00-7419-4fff-b28d-a26f33aeffac/?region=eu-west-2

**Comment by koetsier** [comment]
This: https://github.com/GovWifi/govwifi-smoke-tests/pull/118 makes the smoke tests more robust and fixes the smoke test issues we had. Once merged it should pass
